api/v1alpha1/dnspolicy_types.go

```diff
@@ -70,8 +73,23 @@ type DNSPolicySpec struct {
 
 	// ExcludeAddresses is a list of addresses (either hostnames, CIDR or IPAddresses) that DNSPolicy should not use as values in the configured DNS provider records. The default is to allow all addresses configured in the Gateway DNSPolicy is targeting
 	// +optional
-	// +kubebuilder:validation:MaxItems=20
-	ExcludeAddresses []string `json:"excludeAddresses,omitempty"`
+	ExcludeAddresses ExcludeAddresses `json:"excludeAddresses,omitempty"`
+}
+
+// +kubebuilder:validation:MaxItems=20
+type ExcludeAddresses []string
+
+func (ea ExcludeAddresses) Validate() error {
+	for _, exclude := range ea {
+		//Only a CIDR will have  / in the address so attempt to parse fail if not valid
+		if strings.Contains(exclude, "/") {
+			_, _, err := net.ParseCIDR(exclude)
+			if err != nil {
+				return fmt.Errorf("could not parse the CIDR from the excludeAddresses field %w", err)
+			}
+		}
+	}
+	return nil
 }
```

Comment:
@maleck13 fyi, these are the changes to the v1alpha api file i mentioned this morning. Since this is really a validation issue, adding the validation directly to resource seemed to make sense rather than relying on it failing during reconciliation. Ideally we would be able to do this with CEL validations and prevent the policy ever getting created/updated with invalid values, but now at least we can call this in the validation task.

Reply:
I can copy this change over if it is the only API change we expect

Reply:
There will be additional changes to the API if/when we add section name, but it will be minimal.